passageidentity/passage-ruby
# chore: add dev container

## What's New?



- Adds a dev container config file to bootstrap the dev environment
- Adds rubocop to dev dependencies to satisfy Ruby LSP VS Code extension error when setting rubocop as its formatter
- Replaces openapitools npm package with docker image to codegen because there were more issues than I had time to resolve with putting ruby and node/npm in the same container

**To test:**
1. In VS Code, download the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers)
2. If a popup on your lower right window shows up that asks to **"Reopen in Container"**, select it. Skip to step 5
    
    ![image](https://github.com/user-attachments/assets/72df7d1f-ca92-404e-8cf7-7c4fe88a5384)

3. `cmd + shift + P` to bring up command palette
4. Start typing "rebuild" and select **"Dev Containers: Rebuild and Reopen in Container"**
5. The first time may take a minute or two as it downloads and builds the docker image. Progress can be tracked by clicking the popup

    ![image](https://github.com/user-attachments/assets/663d8790-39b6-4999-978b-ff49b939fc02)

6. Make sure you have your `.env` file populated 
7. Try running linter, formatter, and tests by following commands in the `on-pull-request.yml` CI file or following the documentation for the tools being used
8. Check the extensions and you should find the basics of what you need to dev

**Without VS Code**
If you want to try it without VS Code, you can do so by using the [Dev Container CLI](https://github.com/devcontainers/cli?tab=readme-ov-file#npm-install)

**Codegen change**
codegen change can be tested with
```
curl -o openapi.json https://api.swaggerhub.com/apis/passage/passage-mgmt-sdk-api/1
./generate.sh openapi.json
```

## Screenshots (if appropriate):



## Type of change


## Checklist:


## Additional context
